### PR TITLE
[WFLY-4724] Port SingletonTunnelTestCase from EAP6 to Wildfly

### PR DESCRIPTION
**Pull request by TomasHofman**

### CODE REVIEW

**Comment by rhusar** [comment]
cc @pferraro @rachmatowicz

**Comment by rhusar** [comment]
I have asked Tomas to see if e.g. org.jboss.as.test.clustering.TopologyChangeListener#establishTopology can be reused to cut down SLOC quite a bit and make this more sustainable.

**Comment by pferraro** [comment]
@TomasHofman There are both servlet and remote EJB entry points to this service.  Let me know if its usage is unclear.

**Comment by TomasHofman** [comment]
@rhusar @pferraro I refactored so that it's using already available TopologyChangeListenerServlet. Also, had to make TopologyChangeListenerBean.TIMEOUT configurable.

On my laptop, the test case takes around 2 minutes to finish, which is the reason I put it into ExtendedTests profile. If all looks OK to you, would someone be so kind and trigger CI build with extended tests turned on? It looks I can't do that myself.

**Comment by n1hility** [comment]
@pferraro @rhusar can you approve this latest version?

File: testsuite/integration/clustering/src/test/java/org/jboss/as/test/clustering/tunnel/singleton/SingletonTunnelTestCase.java [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+        return createDeployment();
+    }
+
+    private static GossipRouter gossipRouter = null;
+
+    private static Archive<?> createDeployment() {
+        WebArchive war = ShrinkWrap.create(WebArchive.class, "singleton.war");
+        war.addPackage(MyService.class.getPackage());
+        war.addPackage(SingletonServiceActivator.class.getPackage());
+        war.addClasses(TopologyChangeListener.class, TopologyChangeListenerBean.class, TopologyChangeListenerServlet.class);
```
Comment (pferraro) [inline testsuite/integration/clustering/src/test/java/org/jboss/as/test/clustering/tunnel/singleton/SingletonTunnelTestCase.java]: Please use ClusterTestUtil.addTopologyListenerDependencies(...) instead.

**Comment by TomasHofman** [comment]
@pferraro Done

**Comment by ctomc** [comment]
I've started few more CI jobs for this PR as new clustering tests always make me nervous.

**Comment by rhusar** [comment]
@ctomc Thanks, that wll be great, just checking, did you start them with -DextendedTests since this tests is in that profile? https://github.com/TomasHofman/wildfly/blob/WFLY-4724/testsuite/integration/clustering/pom.xml#L153

**Comment by rhusar** [comment]
This needs a rebase.

**Comment by TomasHofman** [comment]
Rebased.

**Comment by stuartwdouglas** [comment]
@pferraro is this ok to merge now?

**Comment by pferraro** [comment]
:+1: My remaining comments would be purely cosmetic.  This is fine to merge.

**Pull request merged**